api/ttn/lorawan/v3/end_device.proto

```diff
@@ -493,6 +493,56 @@ message ADRSettings {
     }
 
     ChannelSteeringSettings channel_steering = 8;
+
+    // EXPERIMENTAL: Data rate index override settings.
+    message PerDataRateIndexOverride {
+      option (thethings.flags.message) = {
+        select: true,
+        set: true
+      };
+      reserved 1;
+      reserved "min_tx_power_index";
+      reserved 2;
+      reserved "max_tx_power_index";
+      // Minimum number of retransmissions.
+      // If unset, the default value from Network Server configuration will be used.
+      google.protobuf.UInt32Value min_nb_trans = 3 [(validate.rules).uint32 = {
+        gte: 1,
+        lte: 3
+      }];
+      // Maximum number of retransmissions.
+      // If unset, the default value from Network Server configuration will be used.
+      google.protobuf.UInt32Value max_nb_trans = 4 [(validate.rules).uint32 = {
+        gte: 1,
+        lte: 3
+      }];
+    }
+
+    // EXPERIMENTAL: Configuration overrides.
+    message Overrides {
+      option (thethings.flags.message) = {
+        select: true,
+        set: true
+      };
+      PerDataRateIndexOverride data_rate_0 = 1;
```

Comment:
This is modeled like this instead of the obvious `map<DataRateIndex, PerDataRateIndexOverride>` because it aligns better with all of our tooling (CLI parameter generation, field mask generation) and it is also easier to understand.